contracts/Datastore.sol

```diff
@@ -267,24 +315,39 @@ contract Datastore is AragonApp {
     }
 
     /**
-     * Sets IPFS as the storage provider for the datastore.
+     * @notice Change the encryption provider
+     * @param _encryptionProvider Encryption provider
+     */
+    function setEncryptionProvider(EncryptionProvider _encryptionProvider) public {
+        require(settings.encryptionProvider == EncryptionProvider.None);
+        settings.encryptionProvider = _encryptionProvider;
+        emit SettingsChanged(msg.sender);
+    }
+
+    /**
+     * @notice Sets IPFS as the storage provider for the datastore.
      * Since switching between storage providers is not supported,
-     * the method can only be called if storage isn't set or already IPFS
+     * the method can only be called if storage isn't set or already IPFS.
+     * Also sets AES as the encryption provider.
```

Comment:
Why does it choose a fixed encryption provider? Also, see previous comment about already existing files.

Reply:
We have a single encryption provider right now, supporting multiple modes and multiple key length. But maybe in the future we will support more providers. As for the encryption, you are totally right, this was a mistake to have the `|| settings.encryptionProvider == EncryptionProvider.Aes` part. hmm.. but then it would prevent setting the storage provider once the encryption provider is set. We must review this function next week.

Reply:
Ok, I think eventually the storage provider and encryption provider parameters should be more general to be able to accommodate any type of provider, but as for now it's IPFS and AES I guess it's fine, and it can be addressed later.

Reply:
Yeah I agree. We started with the idea of having a general function that accepts a `Settings` struct as parameter but it is not possible right now with Solidity 0.4.24. So we ended up with a function with all the parameters for every providers but it obviously dosn't scale.